### PR TITLE
[docs] Add more detailed docs on repo structure

### PR DESCRIPTION
**Pull request by jameslamb**
As I was working on #2, I realized I was doing too many things at once. Trying to break it up so the PRs are easier to review.

This PR proposes filling in more details on the repo structure, following up on #1 .

1. adds a `CONTRIBUTING.md` with details on the expected structure
2. makes it clear that this repo will not be used to build custom images, and that that should be done in https://github.com/saturncloud/images

Once we agree on this and it's merged, #2 will just be about moving files around to make what we have there compliant, and adding a continuous integration job that will check that all examples follow this structure.

### CODE REVIEW

**Review by rikturr — APPROVED** [review]
love the `.saturn` folder

**Comment by jameslamb** [comment]
> love the `.saturn` folder

thanks! I thought the `.` was a good convention to take from CI services (`.github/`, `.circleci/`) and editors (`.vscode/`, `.idea/`)

**Comment by rikturr** [comment]
> > love the `.saturn` folder
> 
> thanks! I thought the `.` was a good convention to take from CI services (`.github/`, `.circleci/`) and editors (`.vscode/`, `.idea/`)

also added benefit that it doesn't show up in JupyterLab file browser

File: CONTRIBUTING.md
Comment (jsignell) [inline CONTRIBUTING.md:89]: Should this maybe just be included in the json? I don't have a strong preference either way.

File: CONTRIBUTING.md
Comment (jsignell) [inline CONTRIBUTING.md:65]: The project name is just the name of the directory right?

File: README.md
Comment (jsignell) [inline README.md:5]: I think we should just reference contributing from here instead of duplicating content

**Review by jsignell — COMMENTED** [review]
My main comment is to try to avoid duplication in CONTRIBUTING and README

File: CONTRIBUTING.md
Comment (jameslamb) [reply CONTRIBUTING.md:89]: What would the advantage of that be?

I think that the start script should be a script that looks exactly like what you would put in the `Start Script` box in the UI. Otherwise there will be extra mental effort to figure out how the things in the repo relate to the things you see on the "Jupyter" page.

File: README.md
Comment (jameslamb) [reply README.md:5]: ok can do. Let's wait until #6 is agreed and then I'll merge that into this

File: CONTRIBUTING.md
Comment (jameslamb) [reply CONTRIBUTING.md:65]: sure, makes sense

Comment (jsignell) [reply CONTRIBUTING.md:65]: this was agreed in #6

File: CONTRIBUTING.md
Comment (jsignell) [reply CONTRIBUTING.md:89]: The same could be said of environment_variables right?

Comment (jameslamb) [reply CONTRIBUTING.md:89]: it could. The difference with environment variables is that they'll always be exactly one thing: a set of string keys and string values. The order they're defined is also irrelevant, so they don't depend on each other.

Those things aren't true for the `start` script. The `start` script can be any code, and I think that we'd inevitably end up stitching together some parts of it from `saturn.json` plus some parts that are a plain script if we added a `pip_dependencies` section in `saturn.json`.

Comment (jsignell) [reply CONTRIBUTING.md:89]: fair enough :)

**Review by jsignell — APPROVED** [review]
Looks great!

**Comment by jameslamb** [comment]
https://github.com/saturncloud/examples/pull/3/commits/1c8cb87aae73a3b06b9b3edc69152ca96ebdb011 removes the duplicated stuff from `README.md` and pushes it to CONTRIBUTING. It also adds a line explaining that directory names under `examples/` will be project names.

**Pull request merged**
